SOTFEdit/Infrastructure/FlatExtractingYamlLocalizationSource.cs

```diff
@@ -0,0 +1,154 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using libc.translation;
+using YamlDotNet.Serialization;
+
+namespace SOTFEdit.Infrastructure;
+
+/// <summary>
+/// YAML-based localization source that flattens nested YAML structures into dot-notation keys.
+/// Compatible with the existing FlatExtractingJsonLocalizationSource behavior.
+/// </summary>
+public class FlatExtractingYamlLocalizationSource : ILocalizationSource
```

Comment:
Because of this change, the inbuilt translation editor (which saves through TranslationViewModel.Save) will still save to JSON files.

Change it to make it compatible with the new YAML files